types/mercadopago/shared/currency.enum.d.ts

```diff
@@ -0,0 +1,360 @@
+export enum Currency {
```

Comment:
Can you provide location in the source where `Currency` object exists with these properties. I cant seem to find why this should be enum when there does not exist any value named `Currency` anywhere on github in the repo.

Reply:
Sure, let me explain. All `currency_id` in this repo accept only ISO_4217 format and the only way you can figure this out is by looking into the website docs. If you look [here](https://www.mercadopago.com.br/developers/pt/reference/preferences/_checkout_preferences/post/) for example, click the `items` on **Request Params** and it will expand (I know it does not have the arrow-down to indicate that it can be expandable). Inside `items` you can see that it shows the following message: "Identificador de moeda em formato ISO_4217." which translates to "Currency identifier in ISO_4217 format" and ISO_4217 is a 3 letter code standard that represents currencies. You can make sure that it only accept 3 letters by looking at the validation model for all currency_id props, [see this one for example](https://github.com/mercadopago/dx-nodejs/blob/81e3040a77d3a42bb071e67bab4a1907eac4cf51/lib/models/preferencesModel.js#L29). I also want to say that the website is poorly designed and is somewhat incomplete. The SDK is outdated because some properties were included in the api but the SDK has not updated those changes (undocumented properties just give a warning). To write this definition I had to look at the [api website](https://www.mercadopago.com.br/developers/pt/reference/), the [SDK code](https://github.com/mercadopago/dx-nodejs) and the [SDK Wiki](https://github.com/mercadopago/dx-nodejs/wiki). You can see that some properties in the index.d.ts file are `any` and this is because they are simply not in the website so I did not want to wrongly define the types.

Reply:
Also, this enum is just a helper for those using it, so they dont need to look for ISO_4217. Also the website does not say that currency_id is ISO_4217 because it is totally poorly designed. But we can make sure testing (It wil give us an error if you provide something different) and looking into de validation model.